viking/src/tools/check.rs

```diff
@@ -381,6 +391,7 @@ fn check_single(
 fn check_all(checker: &FunctionChecker, functions: &[functions::Info], args: &Args) -> Result<()> {
     let failed = atomic::AtomicBool::new(false);
     let matching_functions: Mutex<HashSet<u64>> = Mutex::new(HashSet::new());
```

Comment:
I don't think we need two separate sets - can we just rename `matching_functions` to something like `new_function_statuses` and turn it into a map of u64 to `functions::Status`? Then you won't need to copy and paste `update_matching_functions`.

Reply:
Done.